frontend/src/Editor/Components/RadioButtonV2/RadioButtonV2.jsx

```diff
@@ -0,0 +1,287 @@
+    setCheckedValue(_value);
+    fireEvent('onSelectionChange');
+  }
+
+  useEffect(() => {
+    if (isInitialRender.current) return;
+    if (advanced) {
+      setCheckedValue(findDefaultItem(schema));
+    } else setCheckedValue(value);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [advanced, JSON.stringify(schema), value]);
+
+  useEffect(() => {
+    if (visibility !== properties.visibility) setVisibility(properties.visibility);
+    if (isLoading !== loadingState) setIsLoading(loadingState);
+    if (isDisabled !== disabledState) setIsDisabled(disabledState);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [properties.visibility, loadingState, disabledState]);
+
+  useEffect(() => {
+    if (isInitialRender.current) return;
+    setExposedVariable('value', checkedValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [checkedValue]);
+
+  useEffect(() => {
+    if (isInitialRender.current) return;
+    const _options = selectOptions?.map(({ label, value }) => ({ label, value }));
+    setExposedVariable('options', _options);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(selectOptions)]);
+
+  useEffect(() => {
+    if (isInitialRender.current) return;
+    setExposedVariable('label', label);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [label]);
+
+  useEffect(() => {
+    if (isInitialRender.current) return;
+    setExposedVariable('isValid', isValid);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isValid]);
+
+  useEffect(() => {
+    if (isInitialRender.current) return;
+    setExposedVariable('isMandatory', isMandatory);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isMandatory]);
+
+  useEffect(() => {
+    if (isInitialRender.current) return;
+    setExposedVariable('isLoading', loadingState);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loadingState]);
+
+  useEffect(() => {
+    if (isInitialRender.current) return;
+    setExposedVariable('isVisible', properties.visibility);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [properties.visibility]);
+
+  useEffect(() => {
+    if (isInitialRender.current) return;
+    setExposedVariable('isDisabled', disabledState);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [disabledState]);
+
+  useEffect(() => {
+    const _options = selectOptions?.map(({ label, value }) => ({ label, value }));
+    const exposedVariables = {
+      value: checkedValue,
+      label: label,
+      options: _options,
+      isValid: isValid,
+      isMandatory: isMandatory,
+      isLoading: loadingState,
+      isVisible: properties.visibility,
+      isDisabled: disabledState,
+      selectOption: async function (value) {
+        onSelect(value);
+      },
+      deselectOption: async function () {
+        setCheckedValue(null);
```

Comment:
just do onSelect(null) try to maintain single source of truth helps keeping the logic simpler